packages/mapviewer/src/store/plugins/geolocation-management.plugin.js

```diff
@@ -27,10 +28,24 @@ function setCenterIfInBounds(store, center) {
 
             if (firstTimeActivatingGeolocation) {
                 firstTimeActivatingGeolocation = !firstTimeActivatingGeolocation
-                store.dispatch('setZoom', {
-                    zoom: store.state.position.projection.get1_25000ZoomLevel(),
-                    ...dispatcher,
-                })
+                if (store.state.cesium.active) {
+                    store.dispatch('setCameraPosition', {
+                        position: {
+                            x: store.state.position.camera.x,
+                            y: store.state.position.camera.y,
+                            z: store.state.position.camera.z,
+
+                            heading: -CesiumMath.toRadians(store.state.position.rotation),
+                            pitch: -90,
+                            roll: 0,
+                        },
+                        dispatcher: self,
+                    })
+                }
+                    store.dispatch('setZoom', {
+                        zoom: store.state.position.projection.get1_25000ZoomLevel(),
+                        ...dispatcher,
+                    })
```

Comment:
not sure the linter worked as intended, you added a `tab` to those lines but these tabs are not needed 🤔